fl4health/server/base_server.py

```diff
@@ -163,6 +164,90 @@ def poll_clients_for_sample_counts(self, timeout: Optional[float]) -> List[int]:
 
         return sample_counts
 
+    def _split_metrics(
+        self, results: List[Tuple[ClientProxy, EvaluateRes]]
+    ) -> Tuple[List[Tuple[ClientProxy, EvaluateRes]], List[Tuple[ClientProxy, EvaluateRes]]]:
+        val_results = []
+        test_results = []
+
+        for client_proxy, eval_res in results:
+            val_metrics = {k: v for k, v in eval_res.metrics.items() if not k.startswith("test - ")}
+            test_metrics = {k: v for k, v in eval_res.metrics.items() if k.startswith("test - ")}
+
+            # Remove loss and num_examples from test_metrics if they exist
+            test_loss = float(test_metrics.pop("test - loss"))
+            test_num_examples = int(test_metrics.pop("test - num_examples"))
+
+            non_test_eval_res = EvaluateRes(eval_res.status, eval_res.loss, eval_res.num_examples, val_metrics)
+            test_eval_res = EvaluateRes(eval_res.status, test_loss, test_num_examples, test_metrics)
+
+            val_results.append((client_proxy, non_test_eval_res))
+            test_results.append((client_proxy, test_eval_res))
+
+        return val_results, test_results
+
+    def _evaluate_round(
+        self,
+        server_round: int,
+        timeout: Optional[float],
+    ) -> Optional[Tuple[Optional[float], Dict[str, Scalar], EvaluateResultsAndFailures]]:
+        """Validate current global model on a number of clients."""
+        # Get clients and their respective instructions from strategy
+        client_instructions = self.strategy.configure_evaluate(
+            server_round=server_round,
+            parameters=self.parameters,
+            client_manager=self._client_manager,
+        )
+        if not client_instructions:
+            log(INFO, "evaluate_round %s: no clients selected, cancel", server_round)
+            return None
+        log(
+            DEBUG,
+            "evaluate_round %s: strategy sampled %s clients (out of %s)",
+            server_round,
+            len(client_instructions),
+            self._client_manager.num_available(),
+        )
+        # Collect `evaluate` results from all clients participating in this round
+        results, failures = evaluate_clients(
+            client_instructions,
+            max_workers=self.max_workers,
+            timeout=timeout,
+        )
+        log(
+            DEBUG,
+            "evaluate_round %s received %s results and %s failures for Validation",
+            server_round,
+            len(results),
+            len(failures),
+        )
+        test_metric_found = any(
```

Comment:
I mocked up a slight simplification using the code for the metric unpacking above for this workflow. Let me know what you think 
```
        val_results, test_results = self._unpack_metrics(results)

        # Aggregate the validation results
        val_aggregated_result: Tuple[
            Optional[float],
            Dict[str, Scalar],
        ] = self.strategy.aggregate_evaluate(server_round, val_results, failures)
        val_loss_aggregated, val_metrics_aggregated = val_aggregated_result

        # Aggregate the test results if they are present
        if len(test_results) > 0:
            test_aggregated_result: Tuple[
                Optional[float],
                Dict[str, Scalar],
            ] = self.strategy.aggregate_evaluate(server_round, test_results, failures)
            test_loss_aggregated, test_metrics_aggregated = test_aggregated_result

            for key, value in test_metrics_aggregated.items():
                val_metrics_aggregated[key] = value
            if test_loss_aggregated is not None:
                val_metrics_aggregated["test - loss - aggregated"] = test_loss_aggregated

        return val_loss_aggregated, val_metrics_aggregated, (results, failures)
```

Reply:
This would allow for you to write a test for `_handle_result_aggregation` to ensure that it works as expected even on edge cases etc.

Reply:
similarly, will add the test- related changes here

Reply:
will do the testing!